.gitignore

```diff
@@ -1,3 +1,8 @@
+bin/
+include/
+src/
+mnist.pkl.gz
+
```

Comment:
The latter should probably come with a comment `# Dataset used in examples`. But what are the first three for?